src/lib/mdns/tests/TestIPCache.cpp

```diff
@@ -0,0 +1,142 @@
+/*
+ *
+ *    Copyright (c) 2021 Project CHIP Authors
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#include <mdns/IPCache.h>
+
+#include <string.h>
+
+#include <cstdint>
+#include <nlunit-test.h>
+#include <support/UnitTestRegistration.h>
+
+#include <core/CHIPError.h>
+#include <core/PeerId.h>
+#include <inet/IPAddress.h>
+#include <inet/InetInterface.h>
+#include <inet/InetLayer.h>
+#include <system/SystemTimer.h>
+#include <system/TimeSource.h>
+
+#include <unistd.h>
+
+using namespace chip;
+using namespace chip::Mdns;
+
+namespace {
+
+#define TEST_IPCACHE_SIZE_SMALL 2
+#define TEST_IPCACHE_TTL_MS 10000
+
+void TestCheckOccupancy(nlTestSuite * inSuite, void * inContext)
+{
+    IPCache<TEST_IPCACHE_SIZE_SMALL, TEST_IPCACHE_TTL_MS> tIPCache;
+
+    NodeId nodeId;
+    FabricId fabricId;
+    uint16_t port;
+    Inet::InterfaceId iface = 0;
+
+    Inet::IPAddress addr;
+    Inet::IPAddress::FromString("1.0.0.1", addr);
```

Comment:
Suggest adding IPv6 test cases and adding TODO comment that the IPv4 test cases are temporary until the support for native IPv6 is complete. Matter is native IPv6 and IPv4 support is not specified. It is currently used a lot by the demos due to the ongoing work taking a long time, but having only IPv4 accounted for in tests avoids being able to keep IPv6 on the line.

Reply:
agreed.   Is there also a way to test mdns with packets on the wire instead of api?